src/openforms/registrations/contrib/objects_api/tests/test_backend.py

```diff
@@ -1191,3 +1208,60 @@ def test_submission_with_objects_api_escapes_html(self, m):
         self.assertEqual(object_create.method, "POST")
         self.assertEqual(object_create.url, "https://objecten.nl/api/v1/objects")
         self.assertEqual(posted_record_data, expected_record_data)
+
+    def test_submission_with_payment(self, m):
+        submission = SubmissionFactory.from_components(
+            [
+                {
+                    "key": "test",
+                    "type": "textfield",
+                },
+            ],
+            registration_success=True,
+            submitted_data={"test": "some test data"},
+            language_code="en",
+            registration_result={
+                "url": "https://objecten.nl/api/v1/objects/111-222-333"
+            },
+        )
+        SubmissionPaymentFactory.create(
+            submission=submission,
+            status=PaymentStatus.started,
+            amount=10,
+            public_order_id="",
+        )
+
+        m.post(
+            "https://objecten.nl/api/v1/objects",
+            status_code=201,
+            json=get_create_json,
+        )
+        m.post(
+            "https://documenten.nl/api/v1/enkelvoudiginformatieobjecten",
+            status_code=201,
+            json=generate_oas_component(
+                "documenten",
+                "schemas/EnkelvoudigInformatieObject",
+                url="https://documenten.nl/api/v1/enkelvoudiginformatieobjecten/1",
+            ),
+        )
+
+        plugin = ObjectsAPIRegistration(PLUGIN_IDENTIFIER)
+        plugin.register_submission(
+            submission,
+            {},
+        )
+
+        self.assertEqual(len(m.request_history), 2)
+
+        object_create = m.last_request
+        body = object_create.json()
+
+        self.assertEqual(
+            body["record"]["data"]["payment"],
+            {
+                "completed": False,
+                "amount": "10,00",
```

Comment:
This is not reliably machine-parseable, it should probably be:

```suggestion
                "amount": 10.00,
```